### PR TITLE
Build tweaks/optimizations

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard1**
## Summary

`@coronadashboard/common` was pulling in all of `date-fns`. Solved that by using named imports. Since that lib is compiled using Typescript we can't use any fancy babel re-writing plugins. Added the `date-fns` babel plugin to the app though, so any usage there will be re-written correctly.

Added `DuplicatePackageCheckerPlugin` to `next.config.js` and resolved a bunch of duplicate packages using Webpack aliases.

Upgraded all of the `visx` packages (they were also responsible for a few duplicates because they were nailed down to a specific version).

Upgraded d3-array and added it to the transpile list.

## Motivation

Trying to fight bundles sizes?.

### Governance

- [ ] Documentation is added
- [ ] Test cases are added or updated
- [ ] I've read the contributing document https://github.com/minvws/.github/blob/master/CONTRIBUTING.md
- [ ] I've read and understand the Code of Conduct https://github.com/minvws/.github/blob/master/CODE_OF_CONDUCT.md
- [ ] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License
      https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/LICENSE.txt
      and the contributor license agreement https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/CLA.md

### CODE REVIEW

**Review by VWSCoronaDashboard14 — APPROVED** [review]

File: packages/app/next.config.js [hunk 1]
```diff
@@ -48,10 +53,36 @@ const nextConfig = {
...
+      ['d3-color', '../../node_modules/d3-interpolate/node_modules/d3-color'],
+      ['d3-geo', '../../node_modules/d3-geo'],
+      ['d3-interpolate', '../../node_modules/d3-interpolate'],
+      ['balanced-match', '../../node_modules/balanced-match'],
+      ['date-fns', 'node_modules/date-fns'],
+    ];
+
+    duplicatePackageResolves.forEach(([packageName, resolvedPath]) => {
+      config.resolve.alias[packageName] = path.resolve(__dirname, resolvedPath);
+    });
```
Comment (VWSCoronaDashboard10) [inline packages/app/next.config.js:76]: this looks a bit like a hack, can't we just fix our dependency versioning? Or use yarn resolutions which exists for these kind of situations?

Comment (VWSCoronaDashboard1) [reply packages/app/next.config.js:76]: This is not a hack, this is one of the recommended ways to solve duplicates:
https://www.npmjs.com/package/duplicate-package-checker-webpack-plugin#resolving-duplicate-packages-in-your-bundle

Comment (VWSCoronaDashboard1) [reply packages/app/next.config.js:76]: As for 'fixing' our dependency versioning, we don't always have 100% control over this. In most cases, these are sub dependencies of dependencies.

Comment (VWSCoronaDashboard10) [reply packages/app/next.config.js:76]: Ah and I see yarn resolutions is also one of the recommended ways. Don't you agree that solution is a bit more elegant?

Comment (VWSCoronaDashboard1) [reply packages/app/next.config.js:76]: I don't have a clear preference, I think both methods have their pros and cons. But if it helps you sleep at night, I don't have a problem with using yarn resolutions. ;)

Comment (VWSCoronaDashboard1) [reply packages/app/next.config.js:76]: I'll make the changes

Comment (VWSCoronaDashboard1) [reply packages/app/next.config.js:76]: Alas, yarn resolutions don't seem to fix the situation, so I'll stick with the 'ugly' solution for now...

**Review by VWSCoronaDashboard8 — APPROVED** [review]

**Pull request merged**